src/eureka/S3_data_reduction/optspex.py

```diff
@@ -511,40 +512,57 @@ def clean_median_flux(data, meta, log, median_thresh=5):
 
     # Create mask using all data quality flags
     mask = np.copy(data['mask'].values)
-    mask[np.where(data.dq > 0)] = 1
+    if meta.dqmask:
+        mask[np.where(data.dq % 2 == 1)] = 0
```

Comment:
No need to do this since it is already done in the main s3_reduce function

Reply:
I had the same thought, and yet it made a difference with/without it when dqmask = True.  Quite unexpected.

Reply:
Oh yikes, that's worth investigating then as that should definitely not be happening

Reply:
I ran some more tests and everything appears to work as expected, so I've now removed the unnecessary code.